fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java

```diff
@@ -946,11 +946,24 @@ private void handleShowVariables() {
     // Show create database
     private void handleShowCreateDb() throws AnalysisException {
         ShowCreateDbStmt showStmt = (ShowCreateDbStmt) stmt;
+        String catalogName = showStmt.getCatalogName();
+        String dbName = showStmt.getDb();
         List<List<String>> rows = Lists.newArrayList();
-        Database db = connectContext.getGlobalStateMgr().getDb(showStmt.getDb());
-        MetaUtils.checkDbNullAndReport(db, showStmt.getDb());
-        rows.add(Lists.newArrayList(showStmt.getDb(),
-                "CREATE DATABASE `" + showStmt.getDb() + "`"));
+
+        if (Strings.isNullOrEmpty(catalogName) || CatalogMgr.isInternalCatalog(catalogName)) {
+            Database db = connectContext.getGlobalStateMgr().getDb(dbName);
+            MetaUtils.checkDbNullAndReport(db, showStmt.getDb());
+            rows.add(Lists.newArrayList(showStmt.getDb(),
+                    "CREATE DATABASE `" + showStmt.getDb() + "`"));
+        } else { // external database
```

Comment:
merge this two block logic. and check if location is null , if it's null, don't display.

Reply:
done